### PR TITLE
feat: active goal quest and notification

### PR DESCRIPTION
**Pull request by shirleyodeng**
## Summary

This PR adds a new quest for active goals and sends a notification to Discord when a goal is accomplished.

## Notion link

https://www.notion.so/talentprotocol/Active-Goal-Quest-Notification-a0c3eedae24e4562a80a3c7d8777e0f2

### CODE REVIEW

File: lib/discord/accomplished_goal_notification.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      @goal = goal
+    end
+
+    def call
+      return unless goal.accomplished?
+      return unless user.present?
+
+      Faraday
+        .post(
+          ENV["DISCORD_WEBHOOK_URL"].to_s,
```
Comment (shirleyodeng) [inline lib/discord/accomplished_goal_notification.rb:13]: Just need this new URL and to change this name once set up on Discord.

File: app/services/quests/refresh_user_quest.rb [hunk 2]
```diff
@@ -157,6 +159,15 @@ def invite_three_quest_completed?
         .count >= 3
     end
 
+    def active_goal_quest_completed?
+      user
```
Comment (RubenSousaDinis) [inline app/services/quests/refresh_user_quest.rb:163]: We have migrated goals to be associated with users yesterday. You can now use `user.goals` directly. We will remove the `career_goal` model soon

File: lib/discord/accomplished_goal_notification.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+          {"Content-Type": "application/json"}
+        )
+    end
+
+    private
+
+    attr_reader :goal
+
+    def user
+      @user ||= goal.career_goal.talent.user
```
Comment (RubenSousaDinis) [inline lib/discord/accomplished_goal_notification.rb:24]:
```suggestion
      @user ||= goal.user
```

File: app/packs/src/components-v2/quest/index.jsx [hunk 4]
```diff
@@ -5,20 +5,21 @@ import { VerifyHumanityQuest } from "./verify-humanity-quest";
 // import { VerifyQuest } from "./verify-quest";
 
 const QUEST_TYPE_MAP = {
+  active_goal: "u/__username__?tab=goals",
```
Comment (RubenSousaDinis) [inline app/packs/src/components-v2/quest/index.jsx:8]: ❤️

**Review by RubenSousaDinis — DISMISSED** [review]
Good job! Let's just solve the conflict and we're ready to go

**Review by RubenSousaDinis — APPROVED** [review]

**Pull request merged**